### ISSUE TITLE
GitLab OAuth broken in Build 5159

### ISSUE THREAD

**Issue by ulope** [mattermost/platform#1875]
When using the current docker `dev` image (Build `5159` `cf0d275ae44ae0e554c039b34a9b68ddac7a081f` according to the logs) OAuth authentication with GitLab is broken.

After having to update the allowed callback URL in GitLab to `mattermost.mydomain.tld/gitlab/complete` despite the documentation sill mentioning `/login/gitlab/complete` and `/signup/gitlab/complete` authentication still breaks after the callback from GitLab with the error message: `Bad token type`.

The log shows this error message: 
```[EROR] /gitlab/complete:AuthorizeOAuthUser code=500 rid=6mh7bigufpyfzjefnqut4ujnmh uid= ip=192.168.99.178 Bad token type [details: token_type=]```

**Comment by coreyhulen** [comment]
RC2 has been released that addresses this issues.  The old routes have been added back.

**Comment by ulope** [comment]
Doesn't appear to be fixed. I've updated the container to the latest version (Build 5188 / b8c70b4d04a7719fd99a8b01485422d1e69dacfd) and still get the same error: `/signup/gitlab/complete:AuthorizeOAuthUser code=500 rid=m639f86me7bj5p3hu6so1ghe4w uid= ip=192.168.99.178 Bad token type [details: token_type=]`

**Comment by coreyhulen** [comment]
hmm...rc2 should be build #5189.  We've tested the build on gitlab.mattermost.com and it appears to be working.

**Comment by ulope** [comment]
Hm maybe the docker tag `dev` uses the wrong build?

**Comment by coreyhulen** [comment]
the `dev` tag is the master branch continuous integration build and may be unstable.  You might want to try `latest` which is `1.3`.  There is also a `1.4` tag that has RC2, but this will be updated soon to the final release of 1.4 by tomorrow.

**Comment by zodman** [comment]
i had the same problem 1.4.0 release

the gitlab log:

```
==> production.log <==
Started POST "/oauth/authorize" for 187.150.204.109 at 2016-01-18 10:51:01 -0600
Processing by Oauth::AuthorizationsController#create as HTML
  Parameters: {"utf8"=>"✓", "authenticity_token"=>"[FILTERED]", "client_id"=>"9a6a48ac206132d3334044af8c5718e061a38925e68a7e68f4d51af8efdbfceb", "redirect_uri"=>"http://zulip.interalia.net/signup/gitlab/complete", "state"=>"eyJhY3Rpb24iOiJsb2dpbiIsImhhc2giOiIkMmEkMTAkV1FlUEZjY3RPMXAuV1E1SGU5QzVlT0JLYzB6Tmp0cGNXc2ZYNkd0WUJmcHJlc0p2TUNFRFciLCJ0ZWFtIjoiaW50ZXJhbGlhIn0=", "response_type"=>"code", "scope"=>"api"}
Redirected to http://zulip.interalia.net/signup/gitlab/complete?code=966c60f87cb6cccd4cf5b3ae84f1cc8b87360a599750953d7711dc5f12c340e1&state=eyJhY3Rpb24iOiJsb2dpbiIsImhhc2giOiIkMmEkMTAkV1FlUEZjY3RPMXAuV1E1SGU5QzVlT0JLYzB6Tmp0cGNXc2ZYNkd0WUJmcHJlc0p2TUNFRFciLCJ0ZWFtIjoiaW50ZXJhbGlhIn0%3D
Completed 302 Found in 89ms (ActiveRecord: 62.8ms)

==> gitlab-workhorse.log <==
git.interalia.net @ - - [2016-01-18 10:51:01.859394982 -0600 CST] "POST /oauth/authorize HTTP/1.1" 302 342 "https://git.interalia.net/oauth/authorize?response_type=code&client_id=9a6a48ac206132d3334044af8c5718e061a38925e68a7e68f4d51af8efdbfceb&redirect_uri=http%3A%2F%2Fzulip.interalia.net%2Fsignup%2Fgitlab%2Fcomplete&state=eyJhY3Rpb24iOiJsb2dpbiIsImhhc2giOiIkMmEkMTAkV1FlUEZjY3RPMXAuV1E1SGU5QzVlT0JLYzB6Tmp0cGNXc2ZYNkd0WUJmcHJlc0p2TUNFRFciLCJ0ZWFtIjoiaW50ZXJhbGlhIn0%3D" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/45.0.2454.85 Safari/537.36" 0.103825

```
i can see the gitlab its not the problem, i cant find the route for: /signup/gitlab/complete on mattermost

**Comment by it33** [comment]
Could we have your help looking at the "About Mattermost" dialog from the team site main menu and let us know if the build number is 5232?

**Comment by zodman** [comment]
i test with master branch and its work ...

**Comment by it33** [comment]
Hi @zodman, hi @ulope, 

Thanks for reporting this, I believe the original issue was addressed (we just shipped and tested Mattermost 2.0 with GitLab 8.5 and everything seemed to work). 

If you're still hitting issues, please help us by opening a new ticket to move this to the top? 

Highly appreciate your reports,
